src/HomeNet/Network/ApplicationServices.cs

```diff
@@ -101,5 +105,84 @@ public HashSet<string> Get()
       return res;
     }
 
+    /// <summary>
+    /// Checks whether the list contains a specific service name.
+    /// </summary>
+    /// <param name="ServiceName">Name of the application service to look for.</param>
+    /// <returns>true if the list contains the specified service name, false otherwise.</returns>
+    public bool ContainsService(string ServiceName)
+    {
+      log.Trace("(ServiceName:'{0}')", ServiceName);
+
+      bool res = false;
+      lock (lockObject)
+      {
+        res = serviceNames.Contains(ServiceName);
+      }
+
+      log.Trace("(-):{0}", res);
+      return res;
+    }
+
+
+    /// <summary>
+    /// Adds a new open relay to the list.
+    /// </summary>
+    /// <param name="Relay">Open relay to add.</param>
+    /// <returns>true if the function succeeds, false if the relay already exists in the list.</returns>
+    public bool AddRelay(RelayConnection Relay)

```

Comment:
node acts as a relay between two clients who wants to communicate, so any messages the client sends to the relay connection is relayed to the other client who reads it